Resources/public/js/views/services/ez-roleserversideviewservice.js

```diff
@@ -0,0 +1,230 @@
+
+            this._assignRoleNotificationStarted(
+                data.roleId, data.roleName, e.selection
+            );
+
+            userService.loadRole(data.roleId, function (error, response) {
+                var role,
+                    tasks = new Y.Parallel(),
+                    end = tasks.add(function (err, response) {
+                        if (err) {
+                            service._error('tasks.add() failed: ' + err.message);
+                            return;
+                        }
+
+                        console.log('end'); // TODO just to let us know that it was picked, error handling should be here
+                    });
+
+                if (error) {
+                    service._error('loadRole failed: ' + error.message);
+                    return;
+                }
+
+                role = response.document.Role;
+
+                Y.Array.each(e.selection, function (struct) {
+                    // TODO assign with limitation
+                    var limitation = null;
+                    //var limitation = { // DUMMY SECTION DATA
+                    //    "_identifier": "Section",
+                    //    "values": {
+                    //        "ref": [{
+                    //            "_href": "/api/ezp/v2/content/sections/1",
+                    //            "_media-type": "application/vnd.ez.api.Section+json"
+                    //        }, {
+                    //            "_href": "/api/ezp/v2/content/sections/4",
+                    //            "_media-type": "application/vnd.ez.api.Section+json"
+                    //        }]
+                    //    }
+                    //};
+                    //var limitation = { // DUMMY SUBTREE DATA
+                    //    "_identifier": "Subtree",
+                    //    "values": {
+                    //        "ref": [{
+                    //            "_href": "/api/ezp/v2/content/locations/1/5/14",
+                    //            "_media-type": "application/vnd.ez.api.Location+json"
+                    //        }]
+                    //    }
+                    //};
+
+                    // Convert hrefs to IDs, required by API
+                    if (limitation) {
+                        for ( var i = 0; i < limitation.values.ref.length; i++ ) {
+                            if (limitation._identifier == "Section") {
+                                limitation.values.ref[i]._href = /\d+$/.exec(limitation.values.ref[i]._href)[0];
+                            } else if (limitation._identifier == "Subtree") {
+                                limitation.values.ref[i]._href = /(\/\d+)+$/.exec(limitation.values.ref[i]._href)[0];
+                            }
+                        }
+                    }
+
+                    var roleAssignInputStruct = userService.newRoleAssignInputStruct(role, limitation),
+                        contentTypeIdentifier = struct.contentType.get('identifier');
```

Comment:
Should be moved to the beginning of that function, just after
`var limitation = null`